foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/oxm/XMLMarshaller.java

```diff
@@ -601,6 +605,7 @@ public void marshal(Object object, MarshalRecord marshalRecord) {
      * @param descriptor the XMLDescriptor for the object being marshalled
      */
     protected void marshal(Object object, MarshalRecord marshalRecord, ABSTRACT_SESSION session, DESCRIPTOR descriptor, boolean isXMLRoot) {
+        logMarshall(object);
```

Comment:
log should be called only if corresponding log level is set (ie if logger.isLoggable(level....) - or sth like that)